BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -408,6 +408,8 @@ private extension ConvivaAnalytics {
     private func buildDynamicContentMetadata() {
         if !player.isLive, player.duration.isFinite {
             contentMetadataBuilder.duration = Int(player.duration)
+        } else {
+            contentMetadataBuilder.duration = nil
```

Comment:
This is needed to avoid keeping the initial value of `0` in case the player API was accessed before the source was loaded.

Reply:
I can't fully follow the thought behind this 🤔 

First, I think the `player.isLive` can be omitted as it's literally the same check as `duration.isFinite`. And second I don't understand why we reset this to `nil` for a live-stream everytime we call this `buildDynamicContentMetadata` (which happens e.g. on every `TimeChangedEvent`). I understand that the duration will then be recalcuatled (set to `-1`) from the `build()` method, though this feels weird to me. 🤔 

Idea:
what if we always set the duration on the `contentMetadataBuilder` (+ changing it to a `TimeInterval`) and handle the `.isFinite` in the `build` method?

Reply:
improved it in a48806e